base/loading.jl

```diff
@@ -408,7 +408,8 @@ end
 ## generic project & manifest API ##
 
 const project_names = ("JuliaProject.toml", "Project.toml")
-const manifest_names = ("JuliaManifest.toml", "Manifest.toml")
+const manifest_names = ("JuliaManifest$(VERSION.major).$(VERSION.minor).toml", "JuliaManifest.toml",
+                        "Manifest$(VERSION.major).$(VERSION.minor).toml", "Manifest.toml")
```

Comment:
This order probably makes more sense
```suggestion
const manifest_names = ("JuliaManifest$(VERSION.major).$(VERSION.minor).toml", "Manifest$(VERSION.major).$(VERSION.minor).toml",
                        "JuliaManifest.toml", "Manifest.toml")
```

Reply:
This looks great to me. I hope we are not wasting time now with a `stat` call (aka `isfile`), since it is substantially faster (and more accurate) to just try to open the file name.

Reply:
@IanButterworth I think this is resolved by 6c36a8d0729c4cf44ba5f0e95fae738ef1851c44, right?

Reply:
The first issue is resolved, but Jameson's point is worth keeping visible. Currently loading does do the bad thing of `isfile` testing rather than just opening because it happens in different places.